sympy/combinatorics/tests/test_homomorphisms.py

```diff
@@ -21,12 +22,30 @@ def test_homomorphism():
     assert T.is_trivial()
     assert T.kernel().order() == G.order()
 
-    F, a = free_group("a")
-    G = FpGroup(F, [a**8])
+    E, e = free_group("e")
+    G = FpGroup(E, [e**8])
     P = PermutationGroup([Permutation(0, 1, 2, 3), Permutation(0, 2)])
-    T = homomorphism(G, P, [a], [Permutation(0, 1, 2, 3)])
+    T = homomorphism(G, P, [e], [Permutation(0, 1, 2, 3)])
     assert T.image().order() == 4
     assert T(T.invert(Permutation(0, 2)(1, 3))) == Permutation(0, 2)(1, 3)
 
-    T = homomorphism(F, AlternatingGroup(4), F.generators, [c])
-    assert T.invert(c**2) == a**2
+    T = homomorphism(E, AlternatingGroup(4), E.generators, [c])
+    assert T.invert(c**2) == e**-1 #order(c) == 3 so c**2 == c**-1
+
+    # FreeGroup -> FreeGroup
+    T = homomorphism(F, E, [a], [e])
+    assert T(a**-2*b**4*a**2).is_identity
+
+    # FreeGroup -> FpGroup
+    G = FpGroup(F, [a*b*a**-1*b**-1])
+    T = homomorphism(F, G, F.generators, G.generators)
+    assert T.invert(a**-1*b**-1*a**2) == a**-1*b**-1*a**2
```

Comment:
It should be `a*b**-1` if `G._rewriting_system` is confluent. But `make_confluent` can take a while so by default it isn't. As for  `b*a**-1: a**-1*b `and `b**-1*a: a*b**-1`, they should have been added to the dictionary when the rules were initiated. I wonder why they weren't. Will work on it.